README.md

```diff
@@ -103,6 +103,30 @@ To remove the package you just installed using:
 
 **DISCLAIMER:** This issue with module imports via `pytest` was the motivating factor to change the project directory structure. For this technique to work, the 'source' code must live inside (nested) a main directory (usually named the same as project directory name or other suitable representative identifier such as **robot** in this case). The `src` subdirectory was renamed because it made no sense when importing a package module by name like `import src.basestation.Motor`, which has no meaning/place in a module semantic context (`import robot.basestation.Motor` is much more appropriate). Most Python projects do not use a `src` directory unless it's for storing their source code that eventually gets compiled to binary (i.e. such as `.c`, `.h`, etc.. files). Also, `base-station` was renamed to `basestation` because Python no-likey dashes in import statements.
 
+### Setup environment variable loader [[`direnv`](https://github.com/direnv/direnv)]
+This tool will be used to take care of managing loading and unloading of environment variables defined in a `.envrc` file in the root. This avoids polluting the global environment.
+
+See the link in title for instructions on how to install and setup.
+
+### Activate venv script [`activatevenv`]
+This script is to be used whenever the dev environment needs to be installed or activated. It does the following:
+- Checks to make sure the `venv` directory is present and if not tries to install it (i.e. `virtualenv -p $python_exec_path venv`)
+- Activates the virtual environment script (i.e. `source venv/bin/activate`)
+- Installs required python modules specified in `requirements.txt` and `requirements-dev.txt`
+- Loads the environment variables with those defined in `.envrc` using `direnv` tool (i.e. `direnv allow .`)
+
+See `activatevenv` for more details and to see how to pass a custom Python path as an argument.
+
+To use it, in the project root directory, simply source it in your shell as follows:
+```
+source activatevenv
+```
+
+Tip: You can `source activatevenv` whenever you need to:
+- Reinstall modules defined in `requirements-*.txt`
+- Reactivate the `venv` environment
+- Reload newly defined variables in `.envrc` as environment variables
```

Comment:
That's what `direnv allow .` does, and that's to avoid sourcing the exported env vars that will pollute the namespace once you're done working on the project (i.e. navigate outside the projects root). Otherwise, yah, we could do that if direnv isn't installed. I'll put that in the if not found block.